test/unit/metadata.js

```diff
@@ -58,6 +58,19 @@ describe('Image metadata', function () {
     });
   });
 
+  it('JPEG with IPTC/XMP', function (done) {
+    sharp(fixtures.inputJpgWithIptcAndXmp).metadata(function (err, metadata) {
+      if (err) throw err;
+      // IPTC
+      assert.strictEqual('object', typeof metadata.iptc);
+      assert.strictEqual(true, metadata.iptc instanceof Buffer);
+      // XMP
+      assert.strictEqual('object', typeof metadata.xmp);
+      assert.strictEqual(true, metadata.xmp instanceof Buffer);
```

Comment:
XMP is XML so perhaps an assertion that the Buffer starts with `<?xpacket` will be enough here?

Reply:
I've went through specs and it looks like <?xpacket will not be enough... I may be wrong though.

Reply:
For the image I added for tests buffer definitely doesn't start from `<?xpacket`. `exiv2 -pS ...` shows that there are two APP1 markers. One is used by EXIF. The other one looks close to what we want:

```
⌘  exiv2 -pS test/fixtures/Landscape_9.jpg
STRUCTURE OF JPEG FILE: test/fixtures/Landscape_9.jpg
 address | marker       |  length | data
       0 | 0xffd8 SOI
       2 | 0xffe0 APP0  |      16 | JFIF............
      20 | 0xffe1 APP1  |   18835 | Exif..II*...............z......
   18857 | 0xffe1 APP1  |   12497 | http://ns.adobe.com/xap/1.0/.<?x
   31356 | 0xffed APP13 |   18252 | Photoshop 3.0.8BIM..............
   49610 | 0xffe2 APP2  |    2444 | ICC_PROFILE......|lcms.0..mntrRG chunk 1/1
   52056 | 0xffdb DQT   |      67
   52125 | 0xffdb DQT   |      67
   52194 | 0xffc0 SOF0  |      17
   52213 | 0xffc4 DHT   |      28
   52243 | 0xffc4 DHT   |      67
   52312 | 0xffc4 DHT   |      27
   52341 | 0xffc4 DHT   |      49
   52392 | 0xffda SOS
```

So `http://ns.adobe.com/xap/1.0/` buffer header may work for tests. But even with this it looks like `metadata.xmp` buffer is wrong.

Reply:
It seems that byte length of both `xmp` and `iptc` property returned are equal. And content looks similar. Seems like we don't get appropriate buffers.